astropy/table/table.py

```diff
@@ -3274,6 +3274,16 @@ def insert_row(self, index, vals=None, mask=None):
         else:
             raise TypeError("Vals must be an iterable or mapping or None")
 
+        if N == 0 and any(
+            isinstance(column, BaseColumn) and isinstance(v, Quantity)
+            for column, v in zip(self.columns.values(), vals)
+        ):
+            warnings.warn(
```

Comment:
How about we also list the solutions? I.e., perhaps add (maybe only for `QTable`?)
```
If you were hoping to fill a QTable row by row, also initialize the units
before starting, i.e., QTable(names=[...], units=[...])
```

Reply:
Seems like a good idea to me. I'll wait for Tom to reply !

Reply:
This special-case help text would only apply for a `N=0` `QTable`. We could do that.

Reply:
It's not much more work to collect the column names of affected columns and print them in the warning, so let's do that.

Reply:
Thanks ! coming right up !

Reply:
Took me a couple minutes since I had to adapt the test and make sure that edge cases were also covered, but it's up now !